src/server/auth/server/testing/oidc_test.go

```diff
@@ -0,0 +1,105 @@
+package server
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/auth"
+	"github.com/pachyderm/pachyderm/src/client/pkg/require"
+)
+
+// TestOIDCAuthCodeFlow tests that we can configure an OIDC provider and do the
+// auth code flow
+func TestOIDCAuthCodeFlow(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	deleteAll(t)
+	adminClient := getPachClient(t, admin)
+
+	conf := &auth.AuthConfig{
+		LiveConfigVersion: 0,
+		IDProviders: []*auth.IDProvider{&auth.IDProvider{
+			Name:        "idp",
+			Description: "fake IdP for testing",
+			OIDC: &auth.IDProvider_OIDCOptions{
+				Issuer:       "http://dex:32000",
+				ClientID:     "pachyderm",
+				ClientSecret: "notsecret",
+				RedirectURI:  "http://pachd:657/authorization-code/callback",
+			},
+		}},
+	}
+	_, err := adminClient.SetConfiguration(adminClient.Ctx(),
+		&auth.SetConfigurationRequest{Configuration: conf})
+	require.NoError(t, err)
+
+	loginInfo, err := adminClient.GetOIDCLogin(adminClient.Ctx(), &auth.GetOIDCLoginRequest{})
+	require.NoError(t, err)
+
+	// Create an HTTP client that doesn't follow redirects.
+	// We rewrite the host names for each redirect to avoid issues because
+	// pachd is configured to reach dex with kube dns, but the tests might be
+	// outside the cluster.
+	c := &http.Client{}
+	c.CheckRedirect = func(_ *http.Request, via []*http.Request) error {
```

Comment:
Brilliant

Reply:
This was one of the issues @adelelopez and I ran into the first time we worked on OIDC, and I remember trying *so hard* to get the same URL to resolve the same way inside and outside the cluster by overwriting `/etc/hosts` and such. This is way simpler.